src/core/src/DynamicTextureProviderCache.cpp

```diff
@@ -0,0 +1,50 @@
+#include "cesium/omniverse/DynamicTextureProviderCache.h"
+
+namespace cesium::omniverse {
+
+void DynamicTextureProviderCache::insert(
+    const std::string& name,
+    std::unique_ptr<omni::ui::DynamicTextureProvider> dynamicTextureProvider) {
+    _items.insert({name, std::make_shared<DynamicTextureProviderCacheItem>(std::move(dynamicTextureProvider))});
+}
+
+void DynamicTextureProviderCache::addReference(const std::string& name) {
+    DynamicTextureProviderCache::Accessor accessor;
+    auto found = _items.find(accessor, name);
+    if (found) {
+        accessor->second->_referenceCount++;
+    }
+}
+
+void DynamicTextureProviderCache::removeReference(const std::string& name) {
+    bool removeItem = false;
+
+    {
+        // Make sure the accessor goes out of scope and the item is unlocked before removing the item
+        DynamicTextureProviderCache::Accessor accessor;
+        auto found = _items.find(accessor, name);
+        if (found) {
+            auto& referenceCount = accessor->second->_referenceCount;
+            referenceCount--;
+            if (referenceCount == 0) {
```

Comment:
As long as `found` is held it blocks other threads, correct? I'm concerned about if there's a case where we could run into a situation where reference count could go negative.

Reply:
I believe `accessor` acts as a lock guard, so we shouldn't see any race conditions

Reply:
Sounds good. This is good to go once you fix the Windows builds. I'll give it a quick once over when that's done.